src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -310,6 +313,10 @@ class JSONReader
                     throw OPENRAVE_EXCEPTION_FORMAT("Has body with no name in file '%s', so cannot load the scene: %s", _filename%orjson::DumpJson(rTempKinBodyInfo), ORE_InvalidArguments);
                 }
             }
+            if ( _excludeBodyIds.count(pKinBodyInfo->_id) ) {
+                itBodyInfo = envInfo._vBodyInfos.erase(itBodyInfo);
```

Comment:
@superfashi does this work? what if the bodyinfo is used later by a reference?

Reply:
Looks to me that `_ProcessEnvInfoBodies` which uses `_ExpandRapidJSON` already resolves the `referenceUri`

Reply:
use `std::find` instead of `count`. thanks

Reply:
Can we use `set::find`?

Reply:
.find is better; std::find is O(N) for (unordered_)set

Reply:
By the way I'm personally surprised about this comment as I thought .count was (kind of) Mujin standard, eg https://github.com/rdiankov/openrave/blob/production/src/libopenrave/kinbodyjoint.cpp#L2417

Reply:
Well... i'm personally surprised that code made it into openrave